src/index.ts

```diff
@@ -171,6 +176,41 @@ interface LoggerExtras<
   flush(cb?: (err?: Error) => void): void;
 }
 
+/**
+ * Configuration options for customizing log output formatting.
+ *
+ * - `maxObjectDepth`: Maximum property depth of objects being logged. Default: 4.
+ * - `omitFunctions`: Whether to omit function properties from logged objects. Default: false (functions are included as "[Function]").
+ * - `stringLength`: Maximum length for string values in logs. Strings longer than this will be truncated. Default: 512.
+ * - `omitHeaderNames`: An array of header names to omit from logged output.
+ * - `serializers`: Custom serialization logic for specific fields.
+ * - `redact`: Paths or rules for redacting sensitive information from log output.
+ * - `redactText`: Custom function for redacting sensitive information from string values in logs. Receives the original string and a placeholder, and should return the redacted string.
+ * - `formatters`: Custom formatting logic for log entries.
+ * - `serializers`: Custom serialization logic for specific fields.
+ */
+type LogFormattingOptions = FormatterOptions &
+  SerializerOptions & {
+    redact?: pino.LoggerOptions['redact'];
+    formatters?: pino.LoggerOptions['formatters'];
+    serializers?: pino.LoggerOptions['serializers'];
+  };
+
+/**
+ * The default options for log formatting applied when none are provided by the user.
+ *
+ * - `omitFunctions`: Whether to omit function properties from logged objects. Default is `false`, meaning functions are included as `"[Function]"`.
+ * - `maxObjectDepth`: The maximum depth to which objects are serialized in logs. Default is `4`.
+ * - `omitHeaderNames`: An array of header names to omit from logged output. Default is the value of `DEFAULT_OMIT_HEADER_NAMES`.
+ * - `stringLength`: The maximum length for string values in logs. Strings longer than this will be truncated. Default is `512`.
+ */
+const defaultLogFormattingOptions: LogFormattingOptions = {
+  omitFunctions: DEFAULT_OMIT_FUNCTIONS,
+  maxObjectDepth: DEFAULT_MAX_OBJECT_DEPTH,
+  omitHeaderNames: DEFAULT_OMIT_HEADER_NAMES,
+  stringLength: DEFAULT_STRING_LENGTH,
```

Comment:
Then maybe we can define these all here / in a central place rather than them being scattered across different `formatter` / `redact` / `serializer` files.